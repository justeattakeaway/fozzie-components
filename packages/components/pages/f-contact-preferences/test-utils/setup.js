export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
