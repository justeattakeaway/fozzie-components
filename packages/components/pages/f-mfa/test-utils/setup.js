export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
