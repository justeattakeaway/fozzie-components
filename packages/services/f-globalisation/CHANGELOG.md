# Changelog

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

v1.0.0
------------------------------
*December 17, 2020*

### Added
- Support for `dateTimeFormats`, which requires a change in the structure of the tenancy files.


v0.2.0
------------------------------
*November 10, 2020*

### Added
- Provided a wrapper around the vue-i18n module for component tests to import


v0.1.0
------------------------------
*November 9, 2020*

### Added
- Added a Globalisation Mixin which performs all required wiring to integrate vue-i18n
