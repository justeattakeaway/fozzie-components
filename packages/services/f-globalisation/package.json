{
  "name": "@justeat/f-globalisation",
  "description": "Fozzie Globalisation â€“ A utility which wires up vue-i18n within your Smart Component",
  "version": "1.1.0",
  "main": "dist/f-globalisation.umd.min.js",
  "maxBundleSize": "15kB",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/services/f-globalisation",
  "contributors": [
    "Github contributors <https://github.com/justeat/fozzie-components/graphs/contributors>"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:justeat/fozzie-components.git"
  },
  "bugs": {
    "url": "https://github.com/justeat/fozzie-components/issues"
  },
  "license": "Apache-2.0",
  "engines": {
    "node": "^12 || ^14 || ^16"
  },
  "keywords": [
    "fozzie"
  ],
  "scripts": {
    "prepublishOnly": "yarn lint && yarn test && yarn build",
    "build": "vue-cli-service build --target lib --name f-globalisation ./src/index.js",
    "lint": "eslint \"!(dist)/**/*.{js,vue}\"",
    "lint:fix": "yarn lint --fix",
    "test": "jest",
    "ci:test:services": "yarn test"
  },
  "browserslist": [
    "extends @justeat/browserslist-config-fozzie"
  ],
  "dependencies": {
    "vue-i18n": "8.28.2"
  }
}
