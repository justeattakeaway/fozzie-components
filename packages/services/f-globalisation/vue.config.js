module.exports = {
    pluginOptions: {
        lintStyleOnBuild: true
    }
};
