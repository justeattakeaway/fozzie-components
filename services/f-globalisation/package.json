{
  "name": "@justeat/f-globalisation",
  "description": "Fozzie Globalisation â€“ A utility which wires up vue-i18n within your Smart Component",
  "version": "0.2.0",
  "main": "dist/f-globalisation.umd.min.js",
  "files": [
    "dist"
  ],
  "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/f-globalisation",
  "contributors": [
    "Github contributors <https://github.com/justeat/fozzie-components/graphs/contributors>"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:justeat/fozzie-components.git"
  },
  "bugs": {
    "url": "https://github.com/justeat/fozzie-components/issues"
  },
  "license": "Apache-2.0",
  "engines": {
    "node": ">=10.0.0"
  },
  "keywords": [
    "fozzie"
  ],
  "scripts": {
    "prepublishOnly": "yarn lint && yarn test && yarn build",
    "build": "vue-cli-service build --target lib --name f-globalisation ./src/index.js",
    "lint": "vue-cli-service lint",
    "lint:fix": "yarn lint --fix",
    "test": "vue-cli-service test:unit --json"
  },
  "browserslist": [
    "extends @justeat/browserslist-config-fozzie"
  ],
  "dependencies": {
    "vue-i18n": "8.0.0"
  },
  "devDependencies": {
    "@samhammer/vue-cli-plugin-stylelint": "2.0.1",
    "@vue/cli-plugin-babel": "4.4.6",
    "@vue/cli-plugin-eslint": "3.9.2",
    "@vue/cli-plugin-unit-jest": "4.4.6",
    "@vue/test-utils": "1.0.3",
    "vue": "2.6.10"
  }
}
