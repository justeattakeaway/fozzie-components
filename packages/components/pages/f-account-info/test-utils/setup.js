import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
