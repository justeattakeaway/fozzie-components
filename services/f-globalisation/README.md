
<div align="center">
  <h1>f-globalisation</h1>

  <img width="125" alt="Fozzie Bear" src="../../bear.png" />

  <p>A utility which wires up vue-i18n within your Smart Component</p>
</div>

---

[![npm version](https://badge.fury.io/js/%40justeat%2Ff-globalisation.svg)](https://badge.fury.io/js/%40justeat%2Ff-globalisation)
[![CircleCI](https://circleci.com/gh/justeat/fozzie-components.svg?style=svg&circle-token=4c77c1990b98c8e06e01b497bc80f376346f609d)](https://circleci.com/gh/justeat/workflows/fozzie-components)
[![Coverage Status](https://coveralls.io/repos/github/justeat/f-globalisation/badge.svg)](https://coveralls.io/github/justeat/f-globalisation)
[![Known Vulnerabilities](https://snyk.io/test/github/justeat/f-globalisation/badge.svg?targetFile=package.json)](https://snyk.io/test/github/justeat/f-globalisation?targetFile=package.json)


## Usage

1.  Install the module using NPM or Yarn:

    ```bash
    yarn add @justeat/f-globalisation
    ```

2.  Import the Mixin

    F-Globalisation contains a mixin which should be imported into your "Smart Component", for example in F-Checkout import it into the Checkout.vue component as that is the root.

    ```
    import { VueGlobalisationMixin } from '@justeat/f-globalisation';

    export default {
        mixins: [VueGlobalisationMixin]
    }
    ```

3.  Add `tenantConfigs` to your component data

    The mixin will access your localisation by accessing this data property. See F-Checkout for an example if needed. It should expose an import for the localisation files in your component.

    ```
    import tenantConfigs from '../tenants';

    export default {
        mixins: [VueGlobalisationMixin],

        data () {
            return {
                tenantConfigs
            }
        }
    }
    ```

4. Now apply vue-i18n within your components

    Once installed; you should be able to access `$t`, `<i18n>` and `this.$i18n`. The correct locale messages should be loaded automatically and `en-GB` should also be loaded as a fallback.

    vue-i18n is not needed as a dependency, because it is registered by the host application, such as CoreWeb or Storybook and thus exists in context.

    The mixin also exposes a prop; so that you can pass a locale in from a website host where required.
