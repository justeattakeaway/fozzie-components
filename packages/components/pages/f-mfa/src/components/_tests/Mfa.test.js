import { createLocalVue, mount, shallowMount } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import sut from '../Mfa.vue';
import tenantConfigs from '../../tenants';
import {
    REDIRECT_URL_EVENT_NAME
} from '../../constants';

const localVue = createLocalVue();
localVue.use(VueI18n);
const validateUrl = 'https://localhost:8080/mfa/validate';
let wrapper;
let sutMocks;
let sutProps;
let sutData;
let mockPostValidateMfaToken;

// Default to en-GB
const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

// Spy on the logging warn method
const warnLogSpy = jest.fn();
const errorLogSpy = jest.fn();
const logMocks = {
    warn: warnLogSpy,
    error: errorLogSpy
};

// Add Spy on the AccountWebApi.postValidateMfaToken method
jest.mock(
    '../../services/AccountWeb.api',
    () => jest.fn().mockImplementation(() => ({
        postValidateMfaToken: mockPostValidateMfaToken
    }))
);

// Provide a generic mount method to be called by each test which
// can override mocks, props and data within each test if required
const mountSut = async ({
    mocks = sutMocks,
    propsData = sutProps,
    data = sutData,
    shallow = true
} = {}) => {
    const mountFn = shallow ? shallowMount : mount;
    const mock = mountFn(sut, {
        i18n,
        localVue,
        propsData,
        data,
        mocks
    });

    await mock.vm.$nextTick();

    return mock;
};

describe('Mfa', () => {
    beforeEach(() => {
        // Arrange clean valid defaults (can be overridden by each test)
        sutData = () => ({
            otp: '',
            tenantConfigs,
            isSubmitting: false,
            showErrorPage: false
        });
        sutProps = {
            validateUrl,
            code: 'AaBbCcDdEeFfGg-HhIiJjKkLlMmNnOoPp_QqRrSsTtUuVv-WwXxYyZz09',
            email: 'jazz.man@hemail.com',
            returnUrl: '/place/i/came/from'
        };
        sutMocks = {
            $log: logMocks
        };
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should be defined', async () => {
        // Act
        wrapper = await mountSut();

        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    it.each`
        pageName          | selector                    | showErrorPage | showHelpInfo
        ${'verification'} | ${'v-mfa-component'}        | ${false}      | ${false}
        ${'error'}        | ${'v-mfa-error-component'}  | ${true}       | ${false}
        ${'help'}         | ${'v-mfa-help-component'}   | ${false}      | ${true}
        ${'help'}         | ${'v-mfa-help-component'}   | ${true}       | ${true}
        `('should show the $pageName page when showErrorPage is $showErrorPage and showHelpInfo is $showHelpInfo', async ({ selector, showErrorPage, showHelpInfo }) => {
        // Arrange
        wrapper = await mountSut({ shallow: false });

        // Act
        await wrapper.setData({ showErrorPage, showHelpInfo });

        // Assert
        expect(wrapper.find(`[data-test-id="${selector}"]`).exists()).toBeTruthy();
    });

    it.each([
        ['enable', 'populated', 'abc123', undefined],
        ['disable', 'unpopulated', '', 'true']
    ])('should %s the submit button if the otp field is %s', async (_, __, otp, expected) => {
        // Arrange
        wrapper = await mountSut();

        // Act
        await wrapper.setData({ otp });

        // Assert
        const button = wrapper.find('[data-test-id="mfa-submit-button"]');
        expect(button.attributes('disabled')).toBe(expected);
    });

    describe('When calling the initialise() method', () => {
        it('should log success if valid props supplied', async () => {
            // Act
            wrapper = await mountSut();

            // Assert
            expect(warnLogSpy).toHaveBeenCalledTimes(1);
            expect(warnLogSpy).toHaveBeenCalledWith('MFA page loaded successfully');
        });

        it.each([
            ['returnUrl', 'https://www.somebadplace.co.uk?returnUrl=nottheplace/i/came/from'],
            ['email', 'mysite..123@hemail.b'],
            ['code', 'ABCDEabcde012345'], // Too short
            ['code', 'ABCDEFG%abcdefg/+=0123456789'], // Illegal characters
            ['code', 'ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz-0123456789'] // Too long
        ])('should show the error page and log warn if the %s validation fails', async (key, value) => {
            // Arrange
            const propsData = {
                ...sutProps,
                [key]: value
            };

            // Act
            wrapper = await mountSut({ propsData });

            // Assert
            const errorCard = wrapper.find('[data-test-id="v-mfa-error-component"]');
            expect(errorCard.exists()).toBe(true);
            expect(warnLogSpy).toHaveBeenCalledTimes(2);
            expect(warnLogSpy).toHaveBeenCalledWith(`Error validating mfa property '${key}' - Regex Failed`, ['account-pages', 'mfa']);
            expect(warnLogSpy).toHaveBeenCalledWith('Error loading MFA page');
        });

        it.each([
            ['email'],
            ['code']
        ])('should show the error page and log warn if the %s values are missing', async key => {
            // Arrange
            const propsData = { ...sutProps };
            delete propsData[key];

            // Act
            wrapper = await mountSut({ propsData });

            // Assert
            const errorCard = wrapper.find('[data-test-id="v-mfa-error-component"]');
            expect(errorCard.exists()).toBe(true);
            expect(warnLogSpy).toHaveBeenCalledTimes(2);
            expect(warnLogSpy).toHaveBeenCalledWith(`Error validating mfa property '${key}' - Regex Failed`, ['account-pages', 'mfa']);
            expect(warnLogSpy).toHaveBeenCalledWith('Error loading MFA page');
        });

        it('should set the default for returnUrl if not supplied', async () => {
            // Arrange
            const expected = '/';
            const propsData = { ...sutProps };
            delete propsData.returnUrl;

            // Act
            wrapper = await mountSut({ propsData });

            // Assert
            const errorCard = wrapper.find('[data-test-id="v-mfa-error-component"]');
            expect(errorCard.exists()).toBe(false);
            expect(wrapper.vm.returnUrl).toBe(expected);

            expect(warnLogSpy).toHaveBeenCalledTimes(1);
            expect(warnLogSpy).toHaveBeenCalledWith('MFA page loaded successfully');
        });
    });

    describe('When calling the onFormSubmit() method', () => {
        it.each([
            ['show', true],
            ['NOT show', false]
        ])('should %s error alert when hasSubmitError is %s', async (_, hasSubmitError) => {
            // Arrange
            wrapper = await mountSut();

            // Act
            await wrapper.setData({ hasSubmitError });

            // Assert
            const alert = wrapper.find('[data-test-id="mfa-submit-error-alert"]');
            expect(alert.exists()).toBe(hasSubmitError);
        });

        it.each([
            ['show', true],
            ['NOT show', false]
        ])('should %s validation error when showValidationError is %s', async (_, showValidationError) => {
            // Arrange
            wrapper = await mountSut({ shallow: false }); // Deep mount to make sure slot is rendered

            // Act
            await wrapper.setData({ showValidationError });

            // Assert
            const validationError = wrapper.find('[data-test-id="mfa-form-validation-error"]');
            expect(validationError.exists()).toBe(showValidationError);
        });

        describe('And the form data is valid', () => {
            it('should call the postValidateMfaToken() method with the correct params', async () => {
                // Arrange
                mockPostValidateMfaToken = jest.fn(() => {}); // Add default mock to the spy
                const expectedParams = { mfa_token: 'XYZ1234', otp: 'test-otp' }; // eslint-disable-line camelcase
                wrapper = await mountSut({
                    propsData:
                        { ...sutProps, code: 'XYZ1234' }
                });
                await wrapper.setData({ otp: 'test-otp' });

                // Act
                await wrapper.vm.onFormSubmit();

                // Assert
                expect(mockPostValidateMfaToken).toHaveBeenCalledWith(expectedParams);
            });

            describe('And the submitting of the form data was unsuccessful', () => {
                it.each([
                    ['Bad request when submitting MFA', 400],
                    ['Throttled when submitting MFA', 429]
                ])("should log the error '%s' if the response status is '%s'", async (logMessage, status) => {
                    // Arrange
                    const errLogged = new Error('some status error message');
                    errLogged.response = { status };
                    mockPostValidateMfaToken = jest.fn(() => { throw errLogged; }); // Add error mock to the spy
                    wrapper = await mountSut();
                    await wrapper.setData({ otp: 'test-otp' });

                    // Act
                    await wrapper.vm.onFormSubmit();

                    // Assert
                    expect(errorLogSpy).toHaveBeenCalledWith(logMessage, errLogged, ['account-pages', 'mfa']);
                    expect(wrapper.vm.isSubmitting).toBe(false);
                });
            });

            describe('And the submitting of the form data was successful', () => {
                it('should emit correct event if api call successful', async () => {
                    // Arrange
                    mockPostValidateMfaToken = jest.fn(() => {}); // Add default mock to the spy
                    const expectedUrl = '/place/i/came/from';
                    wrapper = await mountSut();
                    await wrapper.setData({ otp: 'test-otp' });

                    // Act
                    await wrapper.vm.onFormSubmit();

                    expect(wrapper.emitted()[REDIRECT_URL_EVENT_NAME][0]).toEqual([expectedUrl]);
                    expect(wrapper.vm.isSubmitting).toBe(false);
                });
            });
        });
    });
});
