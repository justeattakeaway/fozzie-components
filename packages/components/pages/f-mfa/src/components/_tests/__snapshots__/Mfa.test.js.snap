// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mfa When calling the initialise() method should log success if valid props supplied 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "form_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "form_action": "view",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should set the default for returnUrl if not supplied 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "form_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "form_action": "view",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should show the error page and log warn if the code validation fails 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should show the error page and log warn if the code validation fails 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should show the error page and log warn if the code validation fails 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should show the error page and log warn if the code values are missing 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should show the error page and log warn if the email validation fails 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should show the error page and log warn if the email values are missing 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the initialise() method should show the error page and log warn if the returnUrl validation fails 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the onFormSubmit() method And the submitting of the form data was successful should emit correct event if api call successful 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "form_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "form_action": "view",
        },
      },
    ],
    Array [
      Object {
        "EventName": "form_success",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "form_action": "success",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the onFormSubmit() method And the submitting of the form data was unsuccessful should log the error 'Bad request when submitting MFA' if the response status is '400' 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "form_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "form_action": "view",
        },
      },
    ],
    Array [
      Object {
        "EventName": "form_error",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "error_message": "http error status : 400",
          "form_action": "error",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the onFormSubmit() method And the submitting of the form data was unsuccessful should log the error 'Throttled when submitting MFA' if the response status is '429' 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "form_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "form_action": "view",
        },
      },
    ],
    Array [
      Object {
        "EventName": "form_error",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "error_message": "http error status : 429",
          "form_action": "error",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Mfa When calling the onFormSubmit() method should call the postValidateMfaToken() method with the correct params 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "EventName": "dialog_view",
        "Parameters": Object {
          "component_name": "multiFactorAuthenticationbadActor",
          "component_type": "dialog",
          "dialog_action": "view",
          "dialog_type": "info",
        },
      },
    ],
    Array [
      Object {
        "EventName": "form_success",
        "Parameters": Object {
          "component_name": "multiFactorAuthentication",
          "component_type": "form",
          "form_action": "success",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
